### PR TITLE
p910nd: +kmod-usb-printer dependency

### PR DESCRIPTION
**Pull request by systemcrash**
Maintainer: @ptpt52 
Run tested: ath79
Description:

It is almost a given that printers on OpenWRT on small routers are connected via USB.

### CODE REVIEW

**Comment by systemcrash** [comment]
We can also ask: how many openwrt routers have a parallel port?

**Comment by BKPepe** [comment]
Yes, we can, but if you noticed that your USB port on your device is not working while plugging the printer, then you have a bigger problem, because first you expect that USB ports are working on the router OOTB. I experienced it once with Xiaomi router, where I flashed OpenWrt and wanted it use it as mass storage. Because of saving 20 kB, I spent 5-10 minutes looking, why it is not working.  Then I argue about it on OpenWrt devel IRC channel and they told me that it is just fine. We dont care about user experience, but about storage, we targets to be minimalistic. It could be possible that someone will not USB port. 🤷 

So, basically from your PR or commit, it is clear that all packages, which are in this repo and requires to be used with USB, should have the dependency, which is not right.

File: net/p910nd/Makefile [hunk 1]
```diff
@@ -28,6 +28,7 @@ define Package/p910nd
   SUBMENU:=Printing
   TITLE:=A small non-spooling printer server
   URL:=http://p910nd.sourceforge.net
+  DEPENDS:=+kmod-usb-printer
```
[the remark this answers is not in the document]
Comment (1715173329) [reply net/p910nd/Makefile:31]: probably: `+USB_SUPPORT:kmod-usb-printer`
I believe size is not a big issue in this case, it would be nice to have a ootb dependency.

**Comment by BKPepe** [comment]
Btw nitpick: Its OpenWrt, not OpenWRT.

File: net/p910nd/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (BKPepe) [reply net/p910nd/Makefile]: Good hint!

**Comment by 1715173329** [comment]
Ping @ptpt52 could you take a look please?

**Comment by Leo-PL** [comment]
Is this dependency really that important? It'll cause woes while trying to install p910nd if you have self-built kernel. Also, p910nd can cover serial-attached printers, too - for example, label printers. Over USB-serial, for example. Also, There exists one USB-parallel converter that works as a (mostly) real port, MCS7715, which isn't covered with kmod-usb-printer. So I believe this is wrong.

**Comment by ptpt52** [comment]
@Leo-PL agree dependency on kmod-usb-printer is not necessary

**Comment by Leo-PL** [comment]
I see yet another problem, this will render the package non-shared. Downloading just `p910nd` for an architecture won't be possible, because the package will depend on exact kernel build.

**Comment by BKPepe** [comment]
In that case, closing.
